components/BacktoTopSection.js

```diff
@@ -0,0 +1,62 @@
+import React, { Component } from 'react'
+import Scroll from 'react-scroll'
+
+export default class TopArrowSection extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      cssHide: 'button-hide'
```

Comment:
`cssHide` is actually a derived state. Right now the code looks like this:

```
           change        bind
handleScroll ==> cssHide <== top arrow
```

This couples the code that handles the scrolling to the CSS class. This makes it harder when we want to add more complex logic to our view (e.g. if we also want to hide the button on desktop devices, for example). This can lead to inconsistent UI state.

But in React, we try to make our code reactive by storing ‘facts,’ and compute the desired state of the view on `render`.

Let’s try to define the meaning by asking few questions…

1. `cssHide` is the CSS class to apply to the top-arrow.

   What CSS class should we apply?

2. We should apply `'button-hide'` class when button should be hidden. Otherwise, we should not apply this class.

   When should the button be hidden?

3. The button should be hidden when the user is at the top of the page. This fact is controlled by the user.

So in this case, I think this is the data that we should store in React state.

```js
  atTopOfPage: false
```

Then on `render`, we can then do this:

```js
  return <div
    id='top-arrow'
    className={`top-arrow ${this.state.atTopOfPage ? 'button-hide' : ''}`}
    onClick={this.scrollToTop}
  />
```

Now we know what class names could be applied to `top-arrow` by looking at the `div` tag only.